flycheck.el

```diff
@@ -3727,10 +3814,13 @@ non-nil."
   (interactive)
   (unless flycheck-mode
     (user-error "Flycheck mode not enabled"))
-  ;; Create and initialize the error list
-  (unless (get-buffer flycheck-error-list-buffer)
-    (with-current-buffer (get-buffer-create flycheck-error-list-buffer)
-      (flycheck-error-list-mode)))
+  ;; Create the error list and reset the filter
+  (with-current-buffer (get-buffer-create flycheck-error-list-buffer)
+    (unless (eq 'major-mode 'flycheck-error-list-mode)
+      (flycheck-error-list-mode))
+    (set (make-local-variable 'flycheck-error-list-minimum-level)
+         (default-value 'flycheck-error-list-minimum-level)))
```

Comment:
I would not make the variable at this place, and rather rely on the global value as long as the user did not press `f`.